test/framework/src/main/java/org/opensearch/test/telemetry/MockTelemetry.java

```diff
@@ -13,25 +13,46 @@
 import org.opensearch.telemetry.metrics.MetricsTelemetry;
 import org.opensearch.telemetry.tracing.TracingTelemetry;
 import org.opensearch.test.telemetry.tracing.MockTracingTelemetry;
+import org.opensearch.threadpool.ThreadPool;
+
+import java.util.concurrent.TimeUnit;
 
 /**
  * Mock {@link Telemetry} implementation for testing.
  */
 public class MockTelemetry implements Telemetry {
-
-    private final TelemetrySettings settings;
+    private final ThreadPool threadPool;
 
     /**
      * Constructor with settings.
      * @param settings telemetry settings.
      */
     public MockTelemetry(TelemetrySettings settings) {
-        this.settings = settings;
+        this(settings, null);
+    }
+
+    /**
+     * Constructor with settings.
+     * @param settings telemetry settings.
+     * @param threadPool thread pool to watch for termination
+     */
+    public MockTelemetry(TelemetrySettings settings, ThreadPool threadPool) {
+        this.threadPool = threadPool;
     }
 
     @Override
     public TracingTelemetry getTracingTelemetry() {
-        return new MockTracingTelemetry();
+        return new MockTracingTelemetry(() -> {
+            // There could be some asynchronous tasks running that we should await for before the closing
+            // up the tracer instance.
+            if (threadPool != null) {
+                try {
+                    threadPool.awaitTermination(10, TimeUnit.SECONDS);
```

Comment:
Just want to point out that we're making an assumption here that any MockTracingTelemetry instance will be closed when the overall application ThreadPool is shut down as well (otherwise this will block for 10 seconds). Is that a valid assumption?

Relatedly, should we assert on the result here? As it is, if the thread pool does not terminate within 10 seconds this will just silently proceed, I believe.

Reply:
> Is that a valid assumption?

Yes, the tracer is closed in the same code block the closes threadPool

> Relatedly, should we assert on the result here?

That is correct, there is not much we can do here, by and large - this is best effort